gapic-generator/lib/gapic/resource_lookup.rb

```diff
@@ -42,7 +42,9 @@ def lookup_file_resource_descriptors file
     def service_resource_types
       @service_resource_types ||= begin
         @service.methods.flat_map do |method|
-          message_resource_types method.input
+          input_resource_types = message_resource_types method.input
+          output_resource_types = message_resource_types method.output
+          input_resource_types.concat output_resource_types
```

Comment:
Better:
```ruby
input_resource_types + output_resource_types
```

Because `.concat` modifies the array in place, which may cause problems if `manage_resource_types` returns one of its internal arrays. (In general, as a rule of thumb, treat Ruby objects as immutable unless you have control over their entire lifecycle, to avoid hard-to-find mutation bugs.)